DEVELOPMENT.md

```diff
@@ -33,91 +34,125 @@ When you submit Pull Requests, `inline-plz` will kick off a [Travis job](blob/ma
+    register_pattern("java", ["*.java"])
+    register_pattern("javascript", ["*.js"])
+    register_pattern("json", ["*.json"])
+    register_pattern("html", ["*.html", "*.htm"])
+    # Add your newlang language and file pattern in alphabetic order if you don't mind
+    register_pattern("newlang", ["*.newlang"])
+    register_pattern("markdown", ["*.md"])
+    register_pattern("python", ["*.py"])
+    register_pattern("shell", ["*.sh", "*.zsh", "*.ksh", "*.bsh", "*.csh", "*.bash"])
+    register_pattern("stylus", ["*.styl"])
+    register_pattern("robotframework", ["*.robot"])
+    register_pattern("rst", ["*.rst"])
+    register_pattern(
+        "text", ["*.md", "*.txt", "*.rtf", "*.html", "*.tex", "*.markdown"]
+    )
+    register_pattern("yaml", ["*.yaml", "*.yml"])
+
 ```
 
 
-## Configure Auto Installation and Create an Execution Configuration
+## Create your Linter class and a unit test
+
+
+Create your linter class in `inlineplz/linters`, then import your linter in [linters/__init__.py](blob/master/inlineplz/linters/__init__.py).
 
-Add configuration for executing your linter [linters/__init__.py](blob/master/inlineplz/linters/__init__.py) (see `LINTERS`).
+Here is the tflint Linter:
 
 ```python
-LINTERS = {
-    "ansible-lint": {
-        "install": [[sys.executable, "-m", "pip", "install", "-U", "ansible-lint"]],
-        "help": ["ansible-lint", "-h"],
-        "run": ["ansible-lint", "-p"],
-        "rundefault": ["ansible-lint", "-p", "-c", "{config_dir}/.ansible-lint"],
-        "dotfiles": [".ansible-lint"],
-        "parser": parsers.AnsibleLintParser,
-        "language": "ansible",
+from __future__ import absolute_import
+from __future__ import unicode_literals
+
+from inlineplz.decorators import linter
+from inlineplz.parsers.base import ParserBase
+import dirtyjson as json
+
+
+@linter(
+    {
+        "name": "tflint",
+        "language": "terraform",
+        "patterns": ["*.tf"],
+        "install": [["brew", "install", "tflint"]],
+        "help": ["tflint", "--help"],
+        "run": ["tflint", "--format=json"],
+        "rundefault": ["tflint", "--format=json"],
+        "dotfiles": [],
         "autorun": True,
-        "run_per_file": True,
-    },
-    ...
-}
+        "run_per_file": False,
+    }
+)
+class TFLintParser(ParserBase):
+    """Parse tflint output."""
+
+
+    def parse(self, lint_data):
+        messages = set()
+
+        for error_stanza in json.loads(lint_data):
+            messages.add(
+                (
+                    error_stanza.get("file", None),
+                    error_stanza.get("line", None),
+                    error_stanza.get("message", None),
+                )
+            )
+
+        return messages
 ```
 
-We'd appreciate it if you kept these in alphabetical order,
+There a few things to point out here:
+
+* When the linter being wrapped supports it we recommed having it produce machine readable output.  We find Json to be *very* convienient.
```

Comment:
[inline-plz]: `proselint: 121: weasel_words.very Substitute 'damn' every time you're inclined to write 'very;' your editor will delete it and the writing will be just as it should be.`